src/approve.ts

```diff
@@ -3,27 +3,34 @@ import * as github from "@actions/github";
 import { RequestError } from "@octokit/request-error";
 import { Context } from "@actions/github/lib/context";
 
-export async function approve(token: string, context: Context) {
-  const { pull_request: pr } = context.payload;
-  if (!pr) {
+export async function approve(
+  token: string,
+  context: Context,
+  pr_number: number
```

Comment:
Would you mind switching this parameter (and its usage below) to camelCase as that's the style used in the rest of the project.

Additionally, I think it'd be better to accept `undefined` for the case that a PR number isn't explicitly provided, rather than using `0` to signal the absence of the value. The `?` after the parameter name achieves that.

```suggestion
  prNumber?: number
```